### PR TITLE
Documentation for other types in collection `requirements.yml`

### PR DESCRIPTION
**Pull request by lilatomic**
##### SUMMARY
Adds documentation on the `file`, `dir`, and `subdirs` types of requirements in a collection's `requirements.yml`

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
`requirements.yml`
`lib/ansible/galaxy/dependency_resolution/dataclasses.py`

### CODE REVIEW

File: docs/docsite/rst/user_guide/collections_using.rst [hunk 1]
```diff
@@ -42,6 +42,11 @@ Installing a collection from a git repository
 
 .. include:: ../shared_snippets/installing_collections_git_repo.txt
 
+Installing a collection from source files
+-----------------------------------------
+
+.. include:: ../shared_snippets/installing_collections_file.rst
+
```
Comment (samccann) [inline docs/docsite/rst/user_guide/collections_using.rst]: Can you make a similar addition to https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/galaxy/user_guide.rst? We cover this info in two places, which is why we use the shared snippets for both of them.

File: docs/docsite/rst/shared_snippets/installing_collections_file.rst [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    collections:
+      # directory containing the collection
+      - source: ./my_namespace/my_collection/
+        type: dir
+
+      # directory containing a namespace, with collections as subdirectories
+      - source: ./my_namespace/
+        type: subdirs
+
+Ansible can also install a collection collected with `ansible-galaxy collection build` by specifying the output file directly:
```
Comment (samccann) [inline docs/docsite/rst/shared_snippets/installing_collections_file.rst]:
```suggestion
Ansible can also install a collection collected with ``ansible-galaxy collection build`` by specifying the output file directly.
```

**Review by samccann — COMMENTED** [review]
@lilatomic sorry for the delay on this PR. It looks like a great addition. I put in a couple of comments to extend this info further.

**Review by s-hertel — APPROVED** [review]

File: docs/docsite/rst/user_guide/collections_using.rst (hunk 1 above)
Comment (samccann) [inline docs/docsite/rst/user_guide/collections_using.rst]: Hey @lilatomic sorry I didn't notice this before. Can you move this new section below the `Install multiple collections with a requirements file` section please? That way we've introduced the requirements file before talking about your new examples.

Comment (lilatomic) [reply docs/docsite/rst/user_guide/collections_using.rst]: good call, makes sense. I reordered the (applicable) sections in user_guide/collections_using to follow those of galaxy/user_guide, lmk if that's good.

**Review by samccann — APPROVED** [review]

**Comment by samccann** [comment]
Thanks @lilatomic for the Ansible docs fix!

**Pull request merged**